### PR TITLE
typescript added

### PR DESCRIPTION
**Pull request by AmeenShaikh025**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+
+// import banner from "./assets/banner.jpg";
+// const banner =  require("./assets/banner.jpg")
+
+function App() {
```
Comment (muskeinsingh) [inline src/App.tsx:6]: return types here

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      "rootDir": "./src",
+      "outDir": "./build/js/bundle.[contenthash].js",
+      "sourceMap": true,
+      "noImplicitAny": true,
+      "moduleResolution": "node",
+      "strict": true,
+      "esModuleInterop": true,
+      "forceConsistentCasingInFileNames": true,
+      "module": "es6",
+      "target": "es5",
```
Comment (muskeinsingh) [inline tsconfig.json:12]: target could be es6 too

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      "jsx": "react",
+      "noEmit": false,
+      "allowJs": true,
```
Comment (muskeinsingh) [inline tsconfig.json:15]: turn false?

**Review by muskeinsingh — COMMENTED** [review]
minor comments

File: package.json [hunk 4]
```diff
@@ -17,22 +17,32 @@
...
     "babel-loader": "^8.0.0",
     "cross-env": "^7.0.3",
     "css-loader": "^5.0.1",
     "css-minimizer-webpack-plugin": "^1.1.5",
     "file-loader": "^6.2.0",
     "html-loader": "^1.3.2",
     "html-webpack-plugin": "^4.5.0",
+    "jest": "^26.6.3",
     "mini-css-extract-plugin": "^1.3.3",
+    "require": "^2.4.20",
```
Comment (muskeinsingh) [inline package.json:32]: maybe you don't need this

**Review by muskeinsingh — COMMENTED** [review]
one more

File: package.json (hunk 4 above)
Comment (AmeenShaikh025) [reply package.json:32]: yes

File: tsconfig.json (hunk 3 above)
Comment (AmeenShaikh025) [reply tsconfig.json:15]: change allowJs to false?

File: tsconfig.json (hunk 2 above)
Comment (AmeenShaikh025) [reply tsconfig.json:12]: ok

File: src/App.tsx (hunk 1 above)
Comment (AmeenShaikh025) [reply src/App.tsx:6]: using typescript ?

File: tsconfig.json (hunk 3 above)
Comment (muskeinsingh) [reply tsconfig.json:15]: yeah

File: src/App.tsx (hunk 1 above)
Comment (muskeinsingh) [reply src/App.tsx:6]: specify return types for the function

**Pull request merged**